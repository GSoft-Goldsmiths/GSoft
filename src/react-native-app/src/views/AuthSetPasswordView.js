import React from 'react';
import {
  AsyncStorage,
  Keyboard,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { fullPageView, style, } from '../styles/variables';
import PasswordInput from '../components/PasswordInput';

export default class AuthSetPasswordView extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      stage: 'enterPassword',
      password: '',
      reEnterPassword: '',
      message: null,
    };
  }

  _setPasswordAsync = async () => {
    const updatingDate = [
      ['isRegistered', 'true'],
      ['password', this.state.password],
      ['userToken', String(Date.now())],
    ];
    try {
      await AsyncStorage.setItem('isRegistered', 'true');
      await AsyncStorage.setItem('password', this.state.password);
      await AsyncStorage.setItem('userToken', String(Date.now()));
      this.props.navigation.navigate('AddTouchId');

    } catch (error) {
      console.log(error);

    }
  };

  componentWillUpdate(nextProps, nextState) {

    if (nextState.stage === 'done') {
      console.log('is done');
      this._setPasswordAsync();
    }

    // in the set password stage
    if (this.state.stage === 'enterPassword') {
      if (nextState.password.length === 4) {
        this.setState({ stage: 'reEnterPassword' });
      }
    }

    // in the set re-enter password stage
    if (this.state.stage === 'reEnterPassword') {
      if (nextState.reEnterPassword.length === 4) {

        // if both passwords are the same
        if (nextState.reEnterPassword === this.state.password) {
          console.log('passwords are same');
          this.setState({ stage: 'done', reEnterPassword: '' });
        } else {

          // if passwords aren't the same, reset it again.
          this.setState({
            stage: 'enterPassword',
            password: '',
            reEnterPassword: '',
            message: 'Passcodes don\'t matched \:(',
          });
        }
      }
    }


  }

  _handleChangeText = (text) => {
    if (this.state.stage === 'enterPassword') {
      this.setState({ password: text });
    } else {
      this.setState({ reEnterPassword: text });
    }
  };

  render() {

    const { stage, message, password, reEnterPassword } = this.state;

    const title = stage === 'enterPassword'
      ? 'Enter 4-digit passcode'
      : 'Re-enter 4-digit passcode';

    const textInputValue = stage === 'enterPassword'
      ? password : reEnterPassword;

    return (
      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
        <View
          style={{
            ...style.container,
            ...fullPageView.withKeyboardContainer,
          }}>
          <PasswordInput
            title={title}
            imageSource={require('../../assets/images/icons/lock.png')}
            imageStyle={fullPageView.image_sm}
            placeholder="◯ ◯ ◯ ◯"
            handleTextChange={this._handleChangeText}
            value={textInputValue}
            message={this.state.message}
          />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}
