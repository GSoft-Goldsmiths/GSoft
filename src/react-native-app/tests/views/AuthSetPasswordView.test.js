import React from 'react';
import renderer from 'react-test-renderer';

import AuthSetPasswordView from '../../src/views/AuthSetPasswordView';

it('renders without crashing', () => {
  const rendered = renderer.create(<AuthSetPasswordView/>).toJSON();
  expect(rendered).toBeTruthy();
});
