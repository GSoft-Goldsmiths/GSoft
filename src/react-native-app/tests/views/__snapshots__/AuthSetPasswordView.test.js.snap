// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={false}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingHorizontal": 20,
      "paddingVertical": 90,
    }
  }
  testID={undefined}
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/icons/lock.png",
      }
    }
    style={
      Object {
        "height": 110,
        "marginBottom": 10,
        "width": 80,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#0093A8",
        "fontFamily": "Caviar Dreams Bold",
        "fontSize": 20,
        "marginBottom": 15,
        "textAlign": "center",
      }
    }
  >
    Enter a new 4-digit passcode
  </Text>
  <TextInput
    allowFontScaling={true}
    autoFocus={true}
    caretHidden={true}
    keyboardType="numeric"
    maxLength={4}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="◯ ◯ ◯ ◯"
    secureTextEntry={true}
    style={
      Object {
        "fontSize": 20,
        "height": 40,
        "textAlign": "center",
      }
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#C00",
        "textAlign": "center",
      }
    }
  />
</View>
`;
